### PR TITLE
Adds global cluster topology locking mechanism

### PR DESCRIPTION
**Pull request by Jotschi**
### General

* [ ] Added abstract that describes the change
* [ ] Added changelog entry to `/CHANGELOG.adoc`
* [ ] Ensured that the change is covered by tests
* [ ] Ensured that the change is documented in the docs

### On API Changes

* [ ] Checked if the changes are breaking or not
* [ ] Added GraphQL API if applicable
* [ ] Added Elasticsearch mapping if applicable

### CODE REVIEW

File: core/src/main/java/com/gentics/mesh/core/endpoint/tagfamily/TagFamilyCrudHandler.java [hunk 1]
```diff
@@ -9,14 +9,15 @@
...
+import com.gentics.mesh.core.verticle.handler.WriteLockImpl;
 import com.gentics.mesh.graphdb.spi.Database;
 
 @Singleton
 public class TagFamilyCrudHandler extends AbstractCrudHandler<TagFamily, TagFamilyResponse> {
 
 	@Inject
-	public TagFamilyCrudHandler(Database db, HandlerUtilities utils) {
-		super(db, utils);
+	public TagFamilyCrudHandler(Database db, HandlerUtilities utils, WriteLockImpl writeLock) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/tagfamily/TagFamilyCrudHandler.java]: Use Interface

File: core/src/main/java/com/gentics/mesh/core/endpoint/user/UserCrudHandler.java [hunk 2]
```diff
@@ -46,8 +47,8 @@
 	private MeshJWTAuthProvider authProvider;
 
 	@Inject
-	public UserCrudHandler(Database db, BootstrapInitializer boot, HandlerUtilities utils, MeshJWTAuthProvider authProvider) {
-		super(db, utils);
+	public UserCrudHandler(Database db, BootstrapInitializer boot, HandlerUtilities utils, MeshJWTAuthProvider authProvider, WriteLockImpl writeLock) {
```
Comment (npomaroli) [inline core/src/main/java/com/gentics/mesh/core/endpoint/user/UserCrudHandler.java]: Use Interface

File: databases/orientdb/src/main/java/com/gentics/mesh/graphdb/OrientDBDatabase.java [hunk 3]
```diff
@@ -335,9 +336,31 @@ public void reload(Element element) {
 	@Override
 	@Deprecated
 	public Tx tx() {
+		blockingTopologyLockCheck();
```
Comment (npomaroli) [inline databases/orientdb/src/main/java/com/gentics/mesh/graphdb/OrientDBDatabase.java]: Remove this

File: databases/orientdb/src/main/java/com/gentics/mesh/graphdb/OrientDBDatabase.java [hunk 4]
```diff
@@ -335,9 +336,31 @@ public void reload(Element element) {
...
+			while (clusterManager().isClusterTopologyLocked()) {
+				long dur = System.currentTimeMillis() - start;
+				if (dur > lockTimeout) {
+					log.warn("Tx global lock timeout of {" + lockTimeout + "} reached.");
+					break;
+				}
+				try {
+					Thread.sleep(100);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
```
Comment (npomaroli) [inline databases/orientdb/src/main/java/com/gentics/mesh/graphdb/OrientDBDatabase.java]: Use log and break

File: databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/OrientDBClusterManager.java [hunk 5]
```diff
@@ -439,4 +439,12 @@ public OHazelcastPlugin getHazelcastPlugin() {
 		return hazelcastPlugin;
 	}
 
+	public boolean isClusterTopologyLocked() {
```
Comment (npomaroli) [inline databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/OrientDBClusterManager.java]: JavaDoc

File: databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/SERVER_STATUS.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.gentics.mesh.graphdb.cluster;
+
+public enum SERVER_STATUS {
```
Comment (npomaroli) [inline databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/SERVER_STATUS.java]: JavaDoc

File: databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/TopologyEventBridge.java [hunk 7]
```diff
@@ -110,4 +130,35 @@ public boolean isVertxReady() {
 		return boot.get().isVertxReady();
 	}
 
+	public boolean isClusterTopologyLocked() {
```
Comment (npomaroli) [inline databases/orientdb/src/main/java/com/gentics/mesh/graphdb/cluster/TopologyEventBridge.java]: JavaDoc

**Comment by GenticsDev** [comment]
Can't set status; build failed.

**Pull request merged**